src/FakeItEasy/Core/ArgumentValueFormatter.cs

```diff
@@ -74,40 +74,11 @@ private static int GetDistanceFromKnownType(Type comparedType, Type knownType)
         private IArgumentValueFormatter ResolveTypeFormatter(Type forType)
         {
             return
-                (from formatter in this.typeFormatters
-                 where formatter.ForType.IsAssignableFrom(forType)
-                 select formatter)
-                .Min(f => new RangedFormatter(f, GetDistanceFromKnownType(f.ForType, forType)))
-                .Formatter;
-        }
-
-        /// <summary>
-        /// Holds a formatter as well as the distance between a type to be formatted
-        /// and the type for which the formatted is registered.
-        /// </summary>
-        private class RangedFormatter : IComparable<RangedFormatter>
-        {
-            private readonly int distanceFromKnownType;
-
-            public RangedFormatter(IArgumentValueFormatter formatter, int distanceFromKnownType)
-            {
-                this.Formatter = formatter;
-                this.distanceFromKnownType = distanceFromKnownType;
-            }
-
-            public IArgumentValueFormatter Formatter { get; }
-
-            public int CompareTo(RangedFormatter other)
-            {
-                Guard.AgainstNull(other, nameof(other));
-
-                if (other.distanceFromKnownType == this.distanceFromKnownType)
-                {
-                    return other.Formatter.Priority.CompareTo(this.Formatter.Priority);
-                }
-
-                return this.distanceFromKnownType.CompareTo(other.distanceFromKnownType);
-            }
+                this.typeFormatters
+                    .Where(formatter => formatter.ForType.IsAssignableFrom(forType))
+                    .OrderBy(formatter => GetDistanceFromKnownType(formatter.ForType, forType))
+                    .ThenByDescending(formatter => formatter.Priority)
+                    .First();

```

Comment:
I like it!

Reply:
To be honest, it's slightly less efficient, since it requires ordering the sequence, so it's `O(n log n)` instead of `O(n)`. What we really needs here is a `MinBy` method, which IMO should have been in Linq since the beginning. Anyway, I don't think it's a problem, since there are typically very few formatters.

Reply:
Yeah, I spotted the performance change, but like you I figured it didn't matter much due to the presumably small list and the hopefully few times the code is used during a test run.